terminus-ui/cohort-date-range/src/cohort-date-range.component.ts

```diff
@@ -125,24 +133,61 @@ export class TsCohortDateRangeComponent implements OnInit, OnDestroy {
   }
 
   /**
-   * Define a reference to the {@link TsSelectComponent}
+   * Get a reference to the cohort form control
+   *
+   * @internal
+   */
+  public get cohortControl(): FormControl {
+    return this.formGroup.get('cohort') as FormControl;
+  }
+
+  /**
+   * Define a reference to the {@link TsSelectionListComponent}
    *
    * @internal
    */
-  @ViewChild(TsSelectComponent, { static: true })
-  public selectComponent: TsSelectComponent;
+  @ViewChild(TsSelectionListComponent, { static: true })
+  public selectionListComponent: TsSelectionListComponent;
 
   /**
-   * Define whether custom date is allowed
+   * Define whether custom dates are allowed
    */
   @Input()
-  public allowCustomDates = true;
+  public set allowCustomDates(value: boolean) {
+    this._allowCustomDates = value;
+
+    // If this flag is being toggled, set the original cohorts again which will add or remove the custom cohort option as needed.
+    this.cohorts = this.originalCohorts;
+  }
+  public get allowCustomDates(): boolean {
+    return this._allowCustomDates;
+  }
+  private _allowCustomDates = true;
 
   /**
-   * Date cohorts
+   * Define the array of date cohorts
    */
   @Input()
-  public cohorts!: ReadonlyArray<TsDateCohort>;
+  public set cohorts(value: ReadonlyArray<TsDateCohort>) {
+    if (!value) {
+      return;
+    }
+    this.originalCohorts = value;
+    this._cohorts = value.slice();
+    if (this.allowCustomDates) {
+      this._cohorts.push(this.customDateCohort);
+    }
+    const activeCohort = value.filter(c => c.active);
+    if (activeCohort.length) {
+      this.cohortControl.setValue(activeCohort);
+      this.setDateRangeValues(activeCohort[0].range);
+    }
+  }
+  public get cohorts(): ReadonlyArray<TsDateCohort> {
+    return this._cohorts;
+  }
+  private _cohorts: TsDateCohort[];
+  private originalCohorts: ReadonlyArray<TsDateCohort>;
```

Comment:
this `originalCohorts` doesn't need a default value?

Reply:
I don't believe so. It's set anytime the cohorts are set and is only used if the `allowUserInput` setter is called.